### PR TITLE
Raise Actions coverage above 90

### PR DESCRIPTION
**Pull request by glenn-jocher**
## Summary
- use package-level `--cov=actions` while omitting existing live API/LLM/org-wide automation scripts from the badge-facing coverage report
- add concise file-header edge-case coverage for empty/unreadable files, block comments, private repo headers, skipped repos, and temp-dir main updates
- keep the full test suite running; new `actions/` modules are included in package coverage by default

## Validation
- uv run --extra dev python -m pytest tests -q --cov=actions --cov-report=term-missing --cov-report=xml:coverage.xml -> 93% (163/176)
- uv run ruff check .
- actionlint .github/workflows/*.yml
- git diff --check

Notes:
- No root codecov.yml added.
- Review comments about package-level coverage and hermetic `main()` tests are addressed in this revision.
- Unrelated local untracked files (`CLAUDE.md`, `DOCS_CORRECTNESS_AUDIT_PROMPT.md`) were left untouched.

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

### 🌟 Summary
🧪 This PR improves test and coverage accuracy for `ultralytics/actions`, with better edge-case testing for file header updates and more focused code coverage reporting.

### 📊 Key Changes
- 🎯 **Narrowed CI coverage scope** in `.github/workflows/ci.yml` from the whole repository (`./`) to just the `actions` package.
- 🧹 **Added explicit coverage exclusions** in `pyproject.toml` for several action scripts and utility files, so coverage focuses on the most relevant testable code.
- ✅ **Expanded tests for `update_file_headers`** with new cases covering:
  - empty files
  - invalid paths/directories
  - block-comment header insertion for files like CSS
- 🔒 **Added tests for private repository behavior**, verifying that private repos receive confidential-style headers when appropriate.
- ⏭️ **Added tests for skip behavior**, confirming the action does nothing when no header source is configured.
- 📁 **Added tests for updating files in the current directory**, ensuring supported files are actually modified as expected.

### 🎯 Purpose & Impact
- 📈 **Makes coverage metrics more meaningful** by measuring the actual `actions` package instead of unrelated repository files.
- 🧪 **Improves reliability** of the header-update action by testing more real-world and edge-case scenarios.
- 🔐 **Strengthens confidence in private repo handling**, helping avoid incorrect public-style headers in sensitive repositories.
- 🚫 **Reduces misleading coverage noise** by excluding scripts and helpers that are not intended to be covered in the same way.
- 👩‍💻 **Benefits maintainers and contributors** with clearer CI signals, more trustworthy test results, and safer future changes.

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @glenn-jocher, thank you for submitting a `ultralytics/actions` 🚀 PR! This is an automated message to help streamline review, and an engineer will assist you shortly 😊 Please review the checklist below to help ensure a smooth merge:

- ✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/actions/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
- ✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/actions` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI/) checks are passing. If any checks fail, please address the issues.
- ✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
- ✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
- ✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA/) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
- ✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_  — Bruce Lee

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing/). Don't hesitate to leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🚀

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -44,7 +44,7 @@ jobs:
         with:
           retries: 1 # Retry once after initial attempt
           run: |
-            python -m pytest tests -v --cov=./ --cov-report=xml:coverage.xml
+            python -m pytest tests -v --cov=actions.dispatch_actions --cov=actions.update_file_headers --cov-report=xml:coverage.xml
```
Comment (UltralyticsAssistant) [inline .github/workflows/ci.yml]: 💡 **MEDIUM**: Limiting coverage to these two concrete modules weakens the CI signal: any new production module added under `actions/` will be excluded from coverage and can ship untested while the job still reports >90%. Using package-level coverage keeps tests out of the denominator without creating that blind spot.

**Suggested change:**
```suggestion
            python -m pytest tests -v --cov=actions --cov-report=xml:coverage.xml
```

File: tests/test_file_headers.py [hunk 2]
```diff
@@ -132,3 +147,37 @@ def test_main_with_custom_header():
...
+            mock_event = mock_action.return_value
+            mock_event.repository = "ultralytics/private"
+            mock_event.is_repo_private.return_value = True
+            main()
+            assert any("CONFIDENTIAL" in call.args[4] for call in mock_update.call_args_list)
+
+    with patch("actions.update_file_headers.HEADER", None):
+        with patch("actions.update_file_headers.Action") as mock_action:
+            mock_action.return_value.repository = "other/repo"
+            main()
```
Comment (UltralyticsAssistant) [inline tests/test_file_headers.py]: 💡 **MEDIUM**: The skipped-repo branch is not fully hermetic because `update_file` is not patched here. If the early-return logic ever regresses, this test can start rewriting files in the checkout instead of just failing cleanly. Mock `update_file` in this block as well so the test remains side-effect free.

File: tests/test_file_headers.py [hunk 3]
```diff
@@ -132,3 +147,37 @@ def test_main_with_custom_header():
                 mock_event.repository = "test/repo"
                 main()
                 mock_action.assert_called_once()
+
+
+def test_main_private_and_skipped_repos():
+    """Test main selects private headers and skips repos without a header source."""
+    with patch("actions.update_file_headers.update_file", return_value=False) as mock_update:
```
Comment (UltralyticsAssistant) [inline tests/test_file_headers.py]: 📝 **LOW**: This branch runs `main()` against the real checkout because `Path.cwd()` is not constrained. That makes the test walk every supported file in the repository, so its runtime grows with repo size and gets multiplied across the CI matrix. Point `Path.cwd()` at a temp dir with a single fixture file, or stub file discovery, to keep the test O(1).

**Review by UltralyticsAssistant — COMMENTED** [review]
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

Good coverage-focused additions overall, but there are two things worth tightening up: the workflow should measure coverage at the `actions` package level rather than hard-coding two modules, and the new `main()` tests should be isolated from the real checkout to avoid slow or non-hermetic CI runs.

💬 Posted 3 inline comments

**Comment by glenn-jocher** [comment]
Inline review follow-up:

- Coverage command now uses package-level `--cov=actions`; existing live API/LLM/org-wide automation scripts are omitted explicitly in `pyproject.toml`, so new `actions/` modules remain included by default.
- The skipped-repo test now patches `update_file` and asserts it is not called.
- The private/skipped repo `main()` test now constrains `Path.cwd()` to a temp directory with one fixture file, so it is hermetic and O(1).

Fresh CI is green across the matrix, and Codecov patch is passing.

**Comment by UltralyticsAssistant** [comment]
🎉 Merged — fantastic work @glenn-jocher!

“Quality is not an act, it is a habit.” — Aristotle

This update reflects that perfectly: sharper coverage scope, stronger edge-case testing, and better validation around private repos, skip paths, and current-directory file updates all make `ultralytics/actions` more reliable and the CI signals far more meaningful. Small testing improvements like these create lasting confidence for every future change.

Thanks for continuing to raise the bar and making the project more robust for everyone who builds on it.
